### PR TITLE
Advancement dialogue PR

### PR DESCRIPTION
**Pull request by ahandleman**

### CODE REVIEW

**Comment by diabeatz96** [comment]
Copy and pasted my other comment onto here: 

https://github.com/diabeatz96/cain/pull/19#issuecomment-2311344448

**Review by diabeatz96 — APPROVED** [review]

**Comment by diabeatz96** [comment]
Approved looks good, if you want to add my other comment changes before merging go right ahead.

**Comment by ahandleman** [comment]
Should be ready to merge

**Pull request merged**
